ui/apps/platform/src/hooks/useURLParameter.test.tsx

```diff
@@ -142,3 +152,47 @@ test('should read/write scoped complex object in URL parameter without changing
     expect(params.get('oldKey')).toBe('test');
     expect(Array.from(params.entries())).toHaveLength(1);
 });
+
+test('should implement push and replace state for history', async () => {
+    let testHistory;
+    let testLocation;
+
+    const { result } = renderHook(() => useURLParameter('testKey', undefined), {
+        initialProps: {
+            children: [],
+            onRouteRender: ({ history, location }) => {
+                testHistory = history;
+                testLocation = location;
+            },
+            initialIndex: 1,
+            initialEntries: ['/main/dashboard', '/main/clusters?oldKey=test'],
+        },
+        wrapper: Wrapper,
+    });
+
+    // Test that specifying a history action of 'push' adds a new entry onto the history stack
+    act(() => {
+        const [, setParam] = result.current;
+        setParam('testValue', 'push');
+    });
+    expect(testLocation.pathname).toBe('/main/clusters');
+    expect(testLocation.search).toBe('?oldKey=test&testKey=testValue');
+    act(() => {
+        testHistory.goBack();
+    });
+    expect(testLocation.pathname).toBe('/main/clusters');
+    expect(testLocation.search).toBe('?oldKey=test');
+
+    // Test the the default behavior is to replace URL parameters in place, i.e. replace history
```

Comment:
If we do change the default action to `push`, don't forget to flip-flop the tests.